### ISSUE TITLE
Summary support

### ISSUE THREAD

**Issue by GavinNovate** [typora/typora-issues#1581]
I would love to see Typora support the summary feature.Just like [this](https://toolchain.gitbook.com/pages.html).

**Comment by abnerlee** [comment]
You could use

```markdown
<details>
  <summary>Copyright 1999-2014.</summary>
  <p> - by Refsnes Data. All Rights Reserved.</p>
  <p>All content and graphics on <a href="https://typora.io" target="blank">this</a> web site are the property of the company Refsnes Data.</p>
</details>
```

instead now in Typora

**Comment by gt-sun** [comment]
能支持代码段吗？

<details>
<summary>View Code</summary>

```json
{
    "kind" : "Policy",
    "apiVersion" : "v1",
    "predicates" : [
        {"name" : "PodFitsHostPorts"},
        {"name" : "PodFitsResources"},
        {"name" : "NoDiskConflict"},
        {"name" : "NoVolumeZoneConflict"},
        {"name" : "MatchNodeSelector"},
        {"name" : "HostName"}
    ],
    "priorities" : [
        {"name" : "LeastRequestedPriority", "weight" : 1},
        {"name" : "BalancedResourceAllocation", "weight" : 1},
        {"name" : "ServiceSpreadingPriority", "weight" : 1},
        {"name" : "EqualPriority", "weight" : 1}
    ]
}
```

</details>

**Comment by Songkeys** [comment]
@abnerlee 

Could we use markdown syntax inside `<details>` tags?
If not, could this feature be supported?

**Comment by vidavidorra** [comment]
Any update on this feature?

This bit does not render in Typora (`Version 0.9.9.25.3 (2897)`). Note that i've added `\` before the ` marks to ensure the code block continues.

```
## collapsible markdown?

<details>
  <summary>CLICK ME</summary>
  <p>

  #### yes, even hidden code blocks!

  \```python
  print("hello world!")
  \```

  </p>
</details>
```

## Render in typora
![image](https://user-images.githubusercontent.com/12871315/61053642-3946dc80-a3ee-11e9-9115-cbbc82e8b39b.png)

## On github

## collapsible markdown?

<details>
  <summary>CLICK ME</summary>
  <p>

  #### yes, even hidden code blocks!

  ```python
  print("hello world!")
  ```
  </p>
</details>

**Comment by abnerlee** [comment]
relates #1939

**Comment by iTomxy** [comment]
I hope it can automatically create hierarchical folding points based on headings of all levels, just like the one in jupyter notebook.

**Comment by svonjoi** [comment]
collapsible images? typora does not rendering image inside <details> tag

**Comment by iTomxy** [comment]
Not image rendering. I mean something like the code folding in a code editor ([example](https://pic2.zhimg.com/v2-732e97a96bca505a67222d1847ec8e15_b.webp)), but the folding points are the headings, i.e. treating the content within the scope of heading as a code block.

**Comment by svonjoi** [comment]
hehe i understood u. I just was asking by collapsible images functionallity, not about your prev comment. Sorry for not expressing myself

**Comment by Squirreljetpack** [comment]
+1 for something like >! syntax for a collapsible region (stackoverflow)

**Comment by dawnkie** [comment]
> You could use
> 
> ```gfm
> <details>
>   <summary>Copyright 1999-2014.</summary>
>   <p> - by Refsnes Data. All Rights Reserved.</p>
>   <p>All content and graphics on <a href="https://typora.io" target="blank">this</a> web site are the property of the company Refsnes Data.</p>
> </details>
> ```
> 
> instead now in Typora

This is so bad for codes

**Comment by pdevito3** [comment]
+1

**Comment by zhaoqingqing** [comment]
mark code fold

**Comment by tracyqwerty** [comment]
same problem.
```
<details>
<summary>Details</summary>
<p>
In C, a string is essentially an array of characters, but it's typically represented as a pointer to its first element -- a `char *`. So when you have a `char *` variable, it is effectively a string.

The `strcmp` function compares two strings. It expects as arguments two pointers to the beginning of each string (two `char *`). When you call `strcmp(wc1->word, wc2->word)`, you're saying "compare the string that begins at `wc1->word` with the string that begins at `wc2->word`".

If you were to dereference the pointers with `*wc1->word` and `*wc2->word`, you would be providing the `char` values (i.e., the first characters of each string) at those memory locations instead of the pointers themselves, which is not what `strcmp` expects.

In other words, `strcmp` doesn't want to know what the first characters of the strings are -- it wants to know where the strings start. By giving it `char *` arguments, you're providing it with that information.

</p>

</details>
```
<img width="789" alt="image" src="https://github.com/typora/typora-issues/assets/44852831/8227bc1f-daf4-4dcb-ab06-478255e24afb">

**Comment by andrew-manger** [comment]
+1 to implement a Markdown version of collapsable content (no HTML) to allow for easy editing and preview.

**Comment by Pomax** [comment]
Counter-point: Typora claims to support GFM, and this is part of GFM, so instead of implementing a _different_ markdown with detail syntax, maybe after almost a decade (it's been an active request _since 2016_) just support it. Or stop claiming to support GFM.

**Comment by MTDickens** [comment]
Why hasn't typora team fixed this after almost 6 yrs!

**Comment by n00mkrad** [comment]
Typora devs don't give a shit about this feature.
